core/src/main/java/org/mule/runtime/core/privileged/util/attribute/ExpressionAttributeEvaluatorDelegate.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.privileged.util.attribute;
+
+import java.io.InputStream;
+import java.io.Serializable;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+import static java.util.Arrays.asList;
+
+import org.mule.runtime.api.component.location.ComponentLocation;
+import org.mule.runtime.api.el.BindingContext;
+import static org.mule.runtime.api.el.BindingContextUtils.NULL_BINDING_CONTEXT;
+
+import org.mule.runtime.api.metadata.DataType;
+import org.mule.runtime.api.metadata.TypedValue;
+import org.mule.runtime.core.api.el.ExpressionManagerSession;
+import org.mule.runtime.core.api.el.ExtendedExpressionManager;
+import org.mule.runtime.core.api.event.CoreEvent;
+
+/**
+ * {@link AttributeEvaluatorDelegate} implementation that resolves EXPRESSION attribute values.
```

Comment:
Rephrase to something like "that resolves attributes values based on expressions"